script.rb

```diff
@@ -0,0 +1,91 @@
+# frozen_string_literal: true
+
+# enumerable ruby module
+module Enumerable
+  def my_each
+    i = 0
+    length.times do
+      yield(self[i])
+      i += 1
+    end
+  end
+
+  def my_each_with_index
+    i = 0
+    length.times do
+      yield(self[i], i)
+      i += 1
+    end
+  end
```

Comment:
In `my_each_with_index`
- my_each_with_index returns an enumerator if no block is given